mungegithub/mungers/publish_scripts/util.sh

```diff
@@ -19,110 +19,446 @@
+    git reset -q --hard upstream-branch
+
+    local k_base_commit=""
+
+    # create filtered-branch-base before filtering for
+    # - new repos that branch off master
+    # - old repos which continue with the last commit.
+    # The filtering will rewrite it. We then fetch the Kubernetes-commit from that and use the
+    # corresponding commit in our repo (it will be different to the one we get by rewriting because we cherry-pick
+    # and add "sync: ..." commits in between).
+    local extra_filter_branches=""
+    if [ "${new_repo}" = "true" ] && [ "${src_branch}" != master ]; then
+        local k_branch_point_commit=$(git merge-base --octopus upstream-kube/${src_branch} upstream-kube/master)
+        if [ -z "${k_branch_point_commit}" ]; then
+            echo "Couldn't find a branch point of upstream-kube/${src_branch} and upstream-kube/master."
+            return 1
         fi
-        if echo ${commits_change_godeps} | grep -q ${commit} > /dev/null; then
-            echo "branch commit ${commit} changes Godeps.json"
-            cherrypicks_change_godeps="true"
-            break
-        fi
-    done <<< "${commits}"
-
-    git checkout ${currBranch}
-
-    # we must reset Godeps.json to what it looked like in the kube-sync branch
-    # before the first commit that's to be cherry-picked so that any new
-    # Godep.json changes from k8s.io/kubernetes will apply cleanly. Note that 
-    # entries for k8s.io/* will be updated later in the process.
-    if [ "${cherrypicks_change_godeps}" = "true" ] && [ "${new_repo}" = "false" ]; then
-       local cleanGodepJsonCommit=${previousBranchSHA}
-       git checkout ${cleanGodepJsonCommit} Godeps/Godeps.json
-       if git diff --cached --exit-code &>/dev/null; then
-           echo "no need to reset Godeps.json!"
-       else
-           git -c user.name="Kubernetes Publisher" -c user.email="k8s-publish-robot@users.noreply.github.com" commit -m "sync: reset Godeps.json" -- Godeps/Godeps.json
-       fi
-    fi
-
-    echo "commits to be cherry-picked:"
-    echo "${commits}"
-    echo ""
-    
-    while read commitSHA; do
-        if [[ -z "${commitSHA}" ]]; then
-            continue
+        echo "Using branch point ${k_branch_point_commit} for new branch ${dst_branch}."
+        git branch -f filtered-branch-base ${k_branch_point_commit} >/dev/null
+        extra_filter_branches="${extra_filter_branches} filtered-branch-base"
+    elif [ "${new_repo}" = false ]; then
+        if [ -z "${k_base_commit}" ]; then
+            # preserve the value from ./kubernetes-sha if it is set. Otherwise, use the Kubernetes-commit lines in the commit message
+            k_base_commit="$(last-kube-commit ${dst_branch} || true)"
+        fi
+        if [ -z "${k_base_commit}" ]; then
+            echo "Couldn't find a Kubernetes-commit sha in any commit on ${dst_branch}."
+            return 1
+        fi
+        local k_base_merge=$(git-find-merge ${k_base_commit} upstream-kube/${src_branch})
```

Comment:
Is it because of commits like this:

![image](https://user-images.githubusercontent.com/2823529/30187845-4b123eac-93e0-11e7-8da6-95a777c51233.png)

Reply:
Yes, exactly. Those are single commits and we do not know whether there are more due to any manual (emergency) pushes.